src/black/__init__.py

```diff
@@ -6336,7 +6349,13 @@ def parse_ast(src: str) -> Union[ast.AST, ast3.AST, ast27.AST]:
                 return ast3.parse(src, filename, feature_version=feature_version)
             except SyntaxError:
                 continue
-
+    if ast27.__name__ == "ast":
+        raise Exception(
+            f"The requested file: {filename!r} has invalid python 3 syntax.\n"
+            "If you are trying to format Python 2 files please reinstall Black with"
+            " the 'python2' extra.\nSee example #7 here: "
+            "https://pip.pypa.io/en/stable/reference/pip_install/#examples"
```

Comment:
please look a bit higher and you should see that `filename` is hard coded to "unknown", not very useful :)

Also, I retracted my feedback about not doing `python3 -m pip install black` since it's questionable at best, so if you feel like it, you can revert to your original suggestion (`python3 -m pip install typed_ast`). The link to the pip docs is worse IMO. I'm sorry for making you flip back and forth on this.

PS. I love your wording here, my wording of anything always uses too many words